src/main/resources/application-dev.properties

```diff
@@ -0,0 +1,11 @@
+spring.datasource.dev.url=jdbc:h2:mem:testDev
```

Comment:
Minor:

There is also **yml** format for application config. From my experience, it's a more common format than **.properties** files.
https://www.baeldung.com/spring-boot-yaml-vs-properties#yaml-configuration